### PR TITLE
fix(cluster-link): ignore actor heartbeat during initialization

### PR DESCRIPTION
**Pull request by qzhuyan**
Fixes [EMQX-12968](https://emqx.atlassian.net/browse/EMQX-12968)
Release version: v/e5.8

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

[EMQX-12968]: https://emqx.atlassian.net/browse/EMQX-12968?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by id** [comment]
If this fix should go to 5.8.0, you need to target `release-58`.

**Comment by qzhuyan** [comment]
@id  thanks for the remind

File: apps/emqx_cluster_link/src/emqx_cluster_link.erl [hunk 1]
```diff
@@ -234,10 +234,17 @@ update_routes(ClusterName, Actor, RouteOps) ->
...
 actor_heartbeat(ClusterName, Actor) ->
-    Env = #{timestamp => erlang:system_time(millisecond)},
-    ActorSt0 = get_actor_state(ClusterName, Actor),
-    ActorSt = emqx_cluster_link_extrouter:actor_apply_operation(heartbeat, ActorSt0, Env),
-    _ = update_actor_state(ActorSt),
+    case erlang:get(?PD_EXTROUTER_ACTOR_STATE) of
+        undefined ->
+            %% skip, only update when it is initialized
+            %% otherwise will crash the init retires
+            skip;
```
Comment (zmstone) [inline apps/emqx_cluster_link/src/emqx_cluster_link.erl:241]: the caller seems to match `ok`

Comment (zmstone) [inline apps/emqx_cluster_link/src/emqx_cluster_link.erl:241]: nit:
```suggestion
            ok;
```

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-3667)

**Pull request merged**